src/app/modules/category/components/filter/filter.component.ts

```diff
@@ -1,12 +1,110 @@
-import { Component, OnInit } from '@angular/core';
+import {
+    Component,
+    OnDestroy,
+    OnInit,
+    ChangeDetectionStrategy,
+} from '@angular/core';
+import { Store } from '@ngrx/store';
+import { Subject } from 'rxjs';
+import { filter, map, takeUntil } from 'rxjs/operators';
+import { ShareService } from 'src/app/services/share/share.service';
+import { AddToArray } from '../../state/category.models';
+import * as CategorySelectors from '../../state/selectors/category.selectors';
 
 @Component({
     selector: 'app-filter',
     templateUrl: './filter.component.html',
     styleUrls: ['./filter.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class FilterComponent implements OnInit {
-    constructor() {}
+export class FilterComponent implements OnInit, OnDestroy {
+    private readonly unsubscribe$ = new Subject();
+
+    array: AddToArray[] = [];
+    private filterItem: AddToArray;
+    inputValue1 = '';
+    inputValue2 = '';
+    selectValue1 = '';
+    selectValue2 = '';
+
+    readonly selectFilterData$ = this.store
+        .select(CategorySelectors.selectFilterData)
+        .pipe(
+            filter(el => el !== undefined),
+            map(el => el),
+            takeUntil(this.unsubscribe$),
+        );
+
+    readonly isReadyToDisplay$ = this.selectFilterData$.pipe(
+        map(el => !!el),
+        takeUntil(this.unsubscribe$),
+    );
+
+    constructor(
+        private readonly store: Store,
+        private readonly shareService: ShareService,
+    ) {}
 
     ngOnInit(): void {}
+
+    onChecked(value: string, selected: boolean, id: number): void {
+        this.filterItem = {
+            first_filter_key: value,
+            id,
+        };
+        if (selected) {
+            this.array.push(this.filterItem);
+        } else {
+            this.array = this.array.filter(
+                el => el.first_filter_key !== this.filterItem.first_filter_key,
+            );
+        }
+        this.shareService.sendData(this.array);
+    }
+
+    onInput(id: number): void {
+        this.onFilter(id, this.inputValue1, this.inputValue2, null);
```

Comment:
you can pass this data from html

Reply:
Yes I can, but then I should create 2 listeners for getting values from inputs. Because now I use 2 different inputs and one listener. Or create some logic inside one listener to get values. But is it worth it?